### PR TITLE
FINERACT-2060: Accrual reverse replay logic and Handling

### PR DESCRIPTION
**Pull request by marta-jankovics**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-e2e-tests-runner/src/test/resources/features/LoanCharge.feature [hunk 1]
```diff
@@ -679,10 +679,6 @@ Feature: LoanCharge
       | ASSET  | 112601       | Loans Receivable        | 1.0   |        |
       | ASSET  | 112603       | Interest/Fee Receivable | 2.0   |        |
       | INCOME | 404007       | Fee Income              |       | 3.0    |
-    Then Loan Transactions tab has a "ACCRUAL" transaction with date "04 November 2022" which has the following Journal entries:
```
Comment (adamsaghy) [inline fineract-e2e-tests-runner/src/test/resources/features/LoanCharge.feature]: Why to remove this check?

File: fineract-e2e-tests-runner/src/test/resources/features/EMICalculation.feature [hunk 2]
```diff
@@ -3292,8 +3435,7 @@ Feature: EMI calculation and repayment schedule checks for interest bearing loan
       | 14 January 2024  | Accrual          | 0.02   | 0.0       | 0.02     | 0.0  | 0.0       | 0.0          | false    | false    |
       | 15 January 2024  | Repayment        | 0.2    | 0.0       | 0.2      | 0.0  | 0.0       | 100.0        | false    | false    |
 
-#  TODO unskip and check when PS-2076 and PS-2106 is done
-  @Skip @TestRailId:C3252
+  @@TestRailId:C3252
```
Comment (adamsaghy) [inline fineract-e2e-tests-runner/src/test/resources/features/EMICalculation.feature]: There is an extra @

File: fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature [hunk 3]
```diff
@@ -538,10 +538,6 @@ Feature: LoanChargeback
       | LIABILITY | 145023       | Suspense/Clearing account | 250.0 |        |
       | ASSET     | 112601       | Loans Receivable          | 250.0 |        |
       | LIABILITY | 145023       | Suspense/Clearing account |       | 250.0  |
-    Then Loan Transactions tab has a "ACCRUAL" transaction with date "30 March 2023" which has the following Journal entries:
```
Comment (adamsaghy) [inline fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature]: Why to remove this?

File: fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature [hunk 4]
```diff
@@ -1076,10 +1072,6 @@ Feature: LoanChargeback
       | LIABILITY | 145023       | Suspense/Clearing account | 250.0 |        |
       | ASSET     | 112601       | Loans Receivable          | 250.0 |        |
       | LIABILITY | 145023       | Suspense/Clearing account |       | 250.0  |
-    Then Loan Transactions tab has a "ACCRUAL" transaction with date "05 April 2023" which has the following Journal entries:
```
Comment (adamsaghy) [inline fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature]: Why to remove this?

File: fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature [hunk 5]
```diff
@@ -2514,10 +2506,6 @@ Feature: LoanChargeback
       | Type      | Account code | Account name              | Debit | Credit |
       | ASSET     | 112601       | Loans Receivable          |       | 250.0  |
       | LIABILITY | 145023       | Suspense/Clearing account | 250.0 |        |
-    Then Loan Transactions tab has a "ACCRUAL" transaction with date "01 April 2024" which has the following Journal entries:
```
Comment (adamsaghy) [inline fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature:2517]: Why to remove this?

File: fineract-e2e-tests-runner/src/test/resources/features/LoanCharge.feature (hunk 1 above)
Comment (marta-jankovics) [reply fineract-e2e-tests-runner/src/test/resources/features/LoanCharge.feature]: put it back

File: fineract-e2e-tests-runner/src/test/resources/features/EMICalculation.feature (hunk 2 above)
Comment (marta-jankovics) [reply fineract-e2e-tests-runner/src/test/resources/features/EMICalculation.feature]: changed

File: fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature (hunk 3 above)
Comment (marta-jankovics) [reply fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature]: put it back

File: fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature (hunk 4 above)
Comment (marta-jankovics) [reply fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature]: put it back

File: fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature (hunk 5 above)
Comment (marta-jankovics) [reply fineract-e2e-tests-runner/src/test/resources/features/LoanChargeback.feature:2517]: no accrual here, as we do not crate accrual without COB

**Pull request merged**